codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionEncoder.java

```diff
@@ -468,6 +464,9 @@ public boolean write(int allowedBytes) {
                             size == bytesWritten && endOfStream, writePromise);
                 } while (size != bytesWritten && allowedBytes > bytesWritten);
                 size -= bytesWritten;
+                if (size == 0 && endOfStream) {
+                    lifecycleManager.closeLocalSide(stream, promise);
```

Comment:
> the problem is that we have to call complete immediately after a frame is removed from the queue (aka passed to Netty).

How about we rename `complete` to `writeComplete`.  So contract would be 0 or 1 calls to `writeComplete` and 0 or 1 calls to `error`, at least 1 of them must be called, and both may be called (`writeComplete` would be called before `error` in this case)?  Not the most straight forward thing...but at least there is a defined interface without duplication. 

> If the encoder calls it on itself, that's his problem :)

His problem...or our problem to fix? :)